modules/solid_mechanics/src/vectorpostprocessors/AverageSectionValueSampler.C

```diff
@@ -88,11 +97,12 @@ AverageSectionValueSampler::AverageSectionValueSampler(const InputParameters & p
   if (!MooseUtils::absoluteFuzzyEqual(_direction.norm_sq(), 1.0))
     paramError("axis_direction", "Axis direction must be a unit vector.");
 
-  _output_vector.resize(_variables.size() + 1);
+  _output_vector.resize(_variables.size() + 2);
   for (const auto j : make_range(_variables.size()))
     _output_vector[j] = &declareVector(_variables[j]);
   const auto pos_idx = _variables.size();
   _output_vector[pos_idx] = &declareVector("axial_position");
+  _output_vector[pos_idx + 1] = &declareVector("node_count");
 
   for (const auto j : make_range(_variables.size()))
   {
```

Comment:
with multi-system coming up you should check that variable system is the same as the UO system

Reply:
I assume that comment doesn't apply to this line specifically. How do you do that?

Reply:
_sys->hasVariable probably

Reply:
I assume that comment doesn't apply to this line specifically. How do you do that?

Reply:
I added a check of _sys->hasVariable(). Let me know if that's not what you had in mind.